go/vt/withddl/withddl.go

```diff
@@ -52,17 +52,29 @@ func (wd *WithDDL) DDLs() []string {
 
 // Exec executes the query using the supplied function.
 // If there are any schema errors, it applies the DDLs and retries.
+// It takes 2 functions, one to run the query and the other to run the
+// DDL commands. This is generally needed so that different users can be used
+// to run the commands. i.e. AllPrivs user for DDLs and App user for query commands.
+// If only 1 function is provided, then that is the function used to run
+// both the type of queries.
 // Funcs can be any of these types:
 // func(query string) (*sqltypes.Result, error)
 // func(query string, maxrows int) (*sqltypes.Result, error)
 // func(query string, maxrows int, wantfields bool) (*sqltypes.Result, error)
 // func(ctx context.Context, query string, maxrows int, wantfields bool) (*sqltypes.Result, error)
-func (wd *WithDDL) Exec(ctx context.Context, query string, f interface{}) (*sqltypes.Result, error) {
-	exec, err := wd.unify(ctx, f)
+func (wd *WithDDL) Exec(ctx context.Context, query string, fQuery interface{}, fDDL interface{}) (*sqltypes.Result, error) {
+	execQuery, err := wd.unify(ctx, fQuery)
 	if err != nil {
 		return nil, err
 	}
-	qr, err := exec(query)
+	execDDL := execQuery
+	if fDDL != nil {
+		execDDL, err = wd.unify(ctx, fQuery)
+		if err != nil {
+			return nil, err
+		}
+	}
+	qr, err := execQuery(query)
```

Comment:
looks like here fDDL needs to be used inside.
Also, can you add a unit test that passes both functions?

Reply:
Yes, fixed both now